WooCommerce/WooCommerceTests/ViewRelated/Search/Order/OrderSearchUICommandTests.swift

```diff
@@ -0,0 +1,171 @@
+        let keyword = "testKeyword"
+        stores.whenReceivingAction(ofType: OrderAction.self) { (action: OrderAction) in
+            guard case let .searchOrders(_, _, _, _, onCompletion) = action else {
+                return XCTFail("Unexpected action: \(action)")
+            }
+            onCompletion(nil)
+        }
+
+        // Mock order statuses
+        insertOrderStatuses()
+
+        // When
+        waitFor { promise in
+            command.synchronizeModels(siteID: self.siteID, keyword: keyword, pageNumber: 1, pageSize: 20) { success in
+                promise(())
+            }
+        }
+
+        // Then
+        guard let event = analyticsProvider.receivedEvents.first else {
+            XCTFail("Expected an event but found none")
+            return
+        }
+        XCTAssertEqual(event, "orders_list_search")
+
+        guard let eventProperties = analyticsProvider.receivedProperties.first else {
+            XCTFail("Expected event properties but found none")
+            return
+        }
+        XCTAssertEqual(eventProperties["search"] as? String, keyword)
+    }
+
+    func testDidSelectSearchResult() throws {
+        // Given
+        let order = Networking.Order.fake().copy()
+
+        // When
+        let selectedOrder = waitFor { promise in
+            let command = OrderSearchUICommand(siteID: self.siteID, onSelectSearchResult: { order, _ in
+                promise(order)
+            }, storageManager: storageManager)
+            command.didSelectSearchResult(model: order, from: .init(), reloadData: {}, updateActionButton: {})
+        }
+
+        // Then
+        XCTAssertEqual(selectedOrder, order)
+    }
+
+    private func insertOrderStatuses() {
+        let statuses: [OrderStatusEnum] = [.pending, .processing, .onHold, .completed, .cancelled, .failed, .custom("aCustomStatus")]
+        statuses.forEach { status in
+            storageManager.insertOrderStatus(name: status.rawValue)
+        }
+        storageManager.viewStorage.saveIfNeeded()
+    }
+
+    private func waitFor<T>(timeout: TimeInterval = 1, _ completion: (_ promise: @escaping (T) -> Void) -> Void) -> T? {
+        var result: T?
+        let expectation = self.expectation(description: "Waiting for completion")
+        completion {
+            result = $0
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: timeout, handler: nil)
+        return result
+    }
+}
+
+/// Mock Order Statuses Store Manager
+///
+private final class MockOrderStatusesStoresManager: MockStorageManager {
+    fileprivate static let siteID: Int64 = 12345
+
+    /// Inserts an order status
+    ///
+    @discardableResult
+    func insertOrderStatus(name: String) -> StorageOrderStatus {
+        let orderStatus = viewStorage.insertNewObject(ofType: StorageOrderStatus.self)
+        orderStatus.name = name
+        orderStatus.slug = name
+        orderStatus.siteID = MockOrderStatusesStoresManager.siteID
```

Comment:
I think it's most reliable if you save these...
```suggestion
        orderStatus.siteID = MockOrderStatusesStoresManager.siteID
        viewStorage.saveIfNeeded()
```

Reply:
Good tip. Addressed in commit 09efa2e.